internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp

```diff
@@ -114,6 +114,33 @@ ExecPlanNodeVisitor::VectorVisitorImpl(VectorPlanNode& node) {
     ret_ = ret;
 }
 
+void
+ExecPlanNodeVisitor::visit(RetrievePlanNode& node) {
+    assert(!retrieve_ret_.has_value());
```

Comment:
Don't use assert(). Print error log instead.

Reply:
Agree, there are some assert() in segcore, I will change those to print error log in later pr.